packages/react-composites/src/composites/CallComposite/adapter/AzureCommunicationCallAdapter.ts

```diff
@@ -830,6 +843,20 @@ export type AzureCommunicationCallAdapterArgs = {
   options?: AzureCommunicationCallAdapterOptions;
 };
 
+/**
+ * Optional parameters to create {@link AzureCommunicationCallAdapter}
+ *
+ * @beta
+ */
+export type TeamsAdapterOptions = {
+  /**
+   * Use this to fetch profile information which will override data in {@link CallAdapterState} like display name
+   * The onFetchProfile is fetch-and-forget one time action for each user, once a user profile is updated, the value will be cached
+   * and would not be updated again within the lifecycle of adapter.
+   */
+  onFetchProfile?: OnFetchProfileCallback;
```

Comment:
(not for this PR, just a thought)
Should we provide any way to know if the information was populated from the calling sdk already?
E.g. I want to come along and only modify profiles were data doesn't already exist.

An alternative to this is in an example usage we pull out of the call remoteparticipants object by default and only do some fancy graph fetch if its undefined and a teams user

Reply:
That's a good point, we can do (id: string, displayName?: string) => Profile for this case

Let's check this in for beta release first, then I will open a new PR to add this one